### ISSUE TITLE
Add a progress bar to show progress when the contributions.txt file is being downloaded

### ISSUE THREAD

**Issue by joelmoniz** [processing/processing#3160]
It would be nice to have a progress bar to show how much of the contributions.txt file has completed being downloaded. This would particularly be handy if the user has a slow internet connection. Here's a quick mock-up:  
  
![contribs_txt_progress_bar](https://cloud.githubusercontent.com/assets/4526417/6869005/6d9360a0-d4b6-11e4-8a70-76074205ac2e.png)

**Comment by benfry** [comment]
@joelmoniz this would be on your plate, would you like me to incorporate #3161 in the meantime?

**Comment by joelmoniz** [comment]
Yes please. I was actually considering marking this as resolved if #3161 was merged, since I thought it might actually looked neater than the progress bar I had proposed. Could you please let me know if the bar proposed here looks better than the status text, as in #3161? If so, I'd love to add in a bar once #3161 is merged.

**Comment by benfry** [comment]
A bar is better in the long-run, but a lower priority with at least the text version in there.

**Comment by joelmoniz** [comment]
Ah perfect. I shall leave this issue open, and shall work on its implementation. Thanks.

**Comment by benfry** [comment]
Incorporated for 3.0a10 with https://github.com/processing/processing/pull/3319
